src/dtx/dtx_rpc.c

```diff
@@ -363,10 +365,22 @@ dtx_cf_rec_update(struct btr_instance *tins, struct btr_record *rec,
 		  d_iov_t *key, d_iov_t *val)
 {
 	struct dtx_req_rec		*drr;
+	struct dtx_id			*tmp;
 	struct dtx_cf_rec_bundle	*dcrb;
 
 	drr = (struct dtx_req_rec *)umem_off2ptr(&tins->ti_umm, rec->rec_off);
 	dcrb = (struct dtx_cf_rec_bundle *)val->iov_buf;
+	if (drr->drr_count == drr->drr_max) {
```

Comment:
If more than one shards locate on the same vos target, then we only need one DTX entry for that, so any time when we call dtx_cf_rec_update(), we can check whether current DTX ID is the same as "drr->drr_dti[drr->drr_count]", if yes, just skip current update.

I have made a patch for that:
https://github.com/daos-stack/daos/pull/3506